crates/edr_provider/src/logger.rs

```diff
@@ -1,143 +1,177 @@
+        spec_id: edr_eth::SpecId,
+        result: &DebugMineBlockResult<Self::BlockchainError>,
     ) {
+        let _spec_id = spec_id;
+        let _result = result;
     }
 
-    /// Logs a single transaction.
-    pub fn log_single_transaction<BlockchainErrorT>(
-        &self,
-        _block: &dyn SyncBlock<Error = BlockchainErrorT>,
-        _transaction: &SignedTransaction,
-        _code: &Bytecode,
-        _gas_used: u64,
-        _trace: &Trace,
+    fn log_mined_block(
+        &mut self,
+        spec_id: edr_eth::SpecId,
+        results: Vec<DebugMineBlockResult<Self::BlockchainError>>,
     ) {
+        let _spec_id = spec_id;
+        let _results = results;
     }
 
-    /// Logs the currently sent transaction.
-    pub fn log_currently_sent_transaction<BlockchainErrorT>(
-        &self,
-        _block: &dyn SyncBlock<Error = BlockchainErrorT>,
-        _transaction: &SignedTransaction,
-        _code: &Bytecode,
-        _gas_used: u64,
-        _trace: &Trace,
+    fn log_send_transaction(
+        &mut self,
+        spec_id: edr_eth::SpecId,
+        transaction: &ExecutableTransaction,
+        mining_results: Vec<DebugMineBlockResult<Self::BlockchainError>>,
     ) {
+        let _spec_id = spec_id;
+        let _transaction = transaction;
+        let _mining_results = mining_results;
     }
 
-    /// Logs the trace of an `eth_estimateGas` call.
-    pub fn log_estimate_gas_trace(
-        &self,
-        _transaction: &TransactionRequest,
-        _code: &Bytecode,
-        _gas_used: u64,
-        _trace: &Trace,
-        _console_log_messages: Vec<String>,
-    ) {
-    }
-
-    /// Logs the trace of an `eth_call` call.
-    pub fn log_call_trace(
-        &self,
-        _transaction: &TransactionRequest,
-        _code: &Bytecode,
-        _trace: &Trace,
-        _console_log_messages: Vec<String>,
-        _error: Option<ProviderError>,
-    ) {
-    }
-
-    /// Logs a warning about multiple transactions being mined.
-    pub fn log_multiple_transactions_warning(&self) {}
-
-    /// Logs a warning about multiple blocks being mined.
-    pub fn log_multiple_blocks_warning(&self) {}
-
-    /// Logs an empty line.
-    pub fn log_empty_line(&self) {}
-
-    /// Print an error message.
-    pub fn print_error_message(&self, _message: &str) {}
-
-    /// Prints a warning message.
-    pub fn print_warning_message(&self, _message: &str) {}
+    /// Returns the logs of the previous request.
+    fn previous_request_logs(&self) -> Vec<String>;
+
+    /// Returns the raw traces of the previous request, if any.
+    fn previous_request_raw_traces(&self) -> Option<Vec<Trace>>;
+
+    /// Prints the collected logs, which correspond to the method with the
+    /// provided name.
+    ///
+    /// Adds an empty line at the end.
+    fn print_method_logs(&mut self, method: &str, error: Option<&ProviderError>);
+
+    // /// Whethers the logger is printing logs to the CLI.
+    // fn is_printing(&self) -> bool;
```

Comment:
Should the commented out methods be removed?

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/4781/commits/f7e00a469e18553c1fae762b8d48a77708f0e11f